Update reserved connection document (#1779)
vitessio/website: content/en/docs/21.0/reference/query-serving/reserved-conn.md

so that it does not actually do anything with user variables. Instead, it keeps 
the state in the Vitess layer.

In other cases, this approach is not enough, and Vitess can use **reserved connections**.

> Reserved connections maintains a dedicated connection for the `vtgate` session 
> from the relevant `vttablet` to the MySQL server. Reserved connections are used when using 

Reviewer: ```suggestion
A dedicated connection is maintained for the `vtgate` session 
from the relevant `vttablet` to its underlying MySQL server. Reserved connections are used when using 
```

the `vtgate` session from the relevant `vttablet` to the MySQL server. Reserved 
connections are used when changing system variables, using temporary tables, 
or when using MySQL locking functions to acquire advisory locks. In general, it 
is better to use reserved connections sparingly, because they reduce the 
In other cases, this approach is not enough, and Vitess can use **reserved connections**.
Reserved connections maintains a dedicated connection for the `vtgate` session 
from the relevant `vttablet` to the MySQL server. Reserved connections are used when using 
temporary tables, or when using MySQL locking functions to acquire advisory locks. 
In general, it is better to use reserved connections sparingly, because they reduce the 
effectiveness of the `vttablet` connection pooling. This may also reduce, or even 
eliminate, the advantages of using connection pooling between `vttablet` and 
MySQL. As such, take note of the `SET` statements that your application's 
MySQL connector and/or ORM sends to MySQL/`vtgate`. Or if those settings will
result in reserved connections being employed for some/all of the application's

> sessions.

Reviewer: This needs to be updated in all affected versions. Which release did this change in?

Author: There is a flag which can be turn it off, so user can still have the impact of using reserved connection with set statement.
It is now deprecated for v21.

Author: @deepthi should I go ahead and merge?

> The `SET` statement used to cause `reserved connection`. This is now solved with the new connection pool implementation used by vttablet.

Reviewer: ```suggestion
`SET` statements used to cause use of `reserved connections`. This is no longer the case with the new connection pool implementation used by vttablet.
```

> The pool tracks the connection with modified settings and not pin the connection to the client session. 

Reviewer: ```suggestion
The connection pool now tracks connections with modified settings instead of pinning connections to specific client sessions. 
```

> Any client requesting connection with or without settings are provided the connection accordingly.

Reviewer: ```suggestion
Any client requesting a connection with or without settings is provided a connection that has the correct settings.
```
